app/src/main/java/org/mozilla/reference/browser/topsites/ui/TopSitesView.kt

```diff
@@ -45,13 +43,11 @@ class TopSitesView @JvmOverloads constructor(
         }
     }
 
-    fun init(
-        loadUrlUseCase: SessionUseCases.LoadUrlUseCase,
-        tabsUseCases: TabsUseCases,
+    fun initialize(
         historyUseCases: HistoryUseCases,
         browserIcons: BrowserIcons
     ) {
-        presenter = TopSitesPresenter(this, loadUrlUseCase, tabsUseCases, historyUseCases)
+        presenter = TopSitesPresenter(context, this, historyUseCases)
```

Comment:
The extra `context` is not needed as a `View` has already an associated `context`

Reply:
Actually this `View` is an interface declared in `TopSitesPresenter` and is not the android view.